src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -54,5 +54,139 @@ const rpc::PlacementGroupTableData &GcsPlacementGroup::GetPlacementGroupTableDat
   return placement_group_table_data_;
 }
 
+/////////////////////////////////////////////////////////////////////////////////////////
+
+GcsPlacementGroupManager::GcsPlacementGroupManager(
```

Comment:
This is currently not handling any sort of recovery after failure, correct?

Reply:
If the gcs server crash, I can't deal with. But If the resource is not enough, i will return the resource to the raylet.
I have store the placement group information in the redis but i don't load them when the gcs crash now.

Reply:
> This is currently not handling any sort of recovery after failure, correct?

Yes, this PR does not contain fault tolerance.